### PR TITLE
feat(copy-paste): copy error, escalation, message and signal references

### PR DESCRIPTION
**Pull request by philippfromme**
* copy references when copying element
* add referenced root element if it doesn't exist
* do NOT add referenced root element if root element with same ID exists
* works around https://github.com/bpmn-io/bpmn-js/issues/798

Related to camunda/camunda-modeler#1049.
Related to camunda/camunda-modeler#1463.

### CODE REVIEW

**Comment by nikku** [comment]
What is the status here? Can I review it?

**Comment by philippfromme** [comment]
@nikku It's ready.

**Comment by philippfromme** [comment]
One possible improvement would be to remove references to root elements when they're deleted, even though that's not possible at the moment. It would allow us to add the ability to remove root elements to the properties panel without worrying about other elements referencing them.

**Comment by philippfromme** [comment]
> * remove referenced root element once it's not referenced anymore

We decided to not automatically remove root elements.

File: lib/features/modeling/behavior/RootElementReferenceBehavior.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    var events = elementRegistry.filter(function(element) {
+      return hasEventDefinition(element, getReferencingEventDefinitionType(rootElement));
+    });
+
+    return some(events, function(event) {
+      var businessObject = getBusinessObject(event),
+          eventDefinitions = businessObject.get('eventDefinitions'),
+          eventDefinition = eventDefinitions[ 0 ];
+
+      return getRootElementReferenced(eventDefinition) === rootElement;
```
Comment (philippfromme) [inline lib/features/modeling/behavior/RootElementReferenceBehavior.js]: We must compare by ID, not by identity.

**Comment by nikku** [comment]
I user tested it. Works mostly already:

* Copy on existing diagram keeps message ref
* Copy across to a different diagram keeps message ref
* Copy to different diagram with message non-existing => creates message

What does not yet work (maybe this is a separate feature request):

* When changing a throwing message event to a catching message intermediate event => Message ref is dropped. Expected => Message ref remains valid and is copied/moved across. This applies for errors, signals, ... too.

**Comment by philippfromme** [comment]
@nikku Shall I continue to work on this?

**Comment by nikku** [comment]
Let's complete this bug fix with the following feature set:

* No automatic cleanup
* Reference elements by ID
* No special handling / keeping of refs on replace (cf. https://github.com/bpmn-io/bpmn-js/pull/1245#issuecomment-559489819). This is another feature request.

**Comment by philippfromme** [comment]
@nikku Ready!

File: lib/features/modeling/behavior/RootElementReferenceBehavior.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  'eventBus',
+  'injector'
+];
+
+inherits(RootElementReferenceBehavior, CommandInterceptor);
+
+// helpers //////////
+
+function hasAnyEventDefinition(element, types) {
+  if (!isArray) {
```
Comment (nikku) [inline lib/features/modeling/behavior/RootElementReferenceBehavior.js]: This looks broken and probably is.

Comment (philippfromme) [reply lib/features/modeling/behavior/RootElementReferenceBehavior.js]: How did this happen? 🤣

Comment (nikku) [reply lib/features/modeling/behavior/RootElementReferenceBehavior.js]: You wonder how I've spotted this? Looked into code coverage :o).

**Comment by philippfromme** [comment]
🎉

**Comment by falko** [comment]
Great thx!

**Pull request merged**